apis/r/src/Makevars.in

```diff
@@ -1,10 +1,10 @@
 CXX_STD = CXX17
 
 ## We need the TileDB Headers, and for macOS aka Darwin need to set minimum version 10.14 for macOS
-PKG_CPPFLAGS = -I. -I../inst/include/ @tiledb_include@ @cxx17_macos@ -Iinclude
+PKG_CPPFLAGS = -Wno-deprecated-declarations -I. -I../inst/include/ @tiledb_include@ @cxx17_macos@ -Iinclude
 
 ## When this becomes a CRAN package we may have to remove this. For now it keeps the noise down
-PKG_CXX17FLAGS = -Wno-deprecated-declarations
+PKG_CXXFLAGS = -Wno-deprecated-declarations
```

Comment:
That line you can remove.  It is redundant. If you look closely at what R CMD INSTALL does you will see two occurrences.  We were already covered from PKG_CPPFLAGS as the preprocessor options will always be added.

```sh
# .... showing just one line from my box with `ccache`
ccache g++-12  -std=gnu++17 -I"/usr/share/R/include" -DNDEBUG -Wno-deprecated-declarations -I.\
-I../inst/include/ -I/usr/local/include   -Iinclude -I'/usr/local/lib/R/site-library/Rcpp/include'\
-I'/usr/local/lib/R/site-library/RcppSpdlog/include'    -Wno-deprecated-declarations \
-fpic  -g -O3 -Wall -pipe -pedantic   -c RcppExports.cpp -o RcppExports.o
```

Reply:
@eddelbuettel thanks

Let's just go ahead and ignore `src/Makevars.in` for this PR -- I put some things in to try to reduce compiler warnings while developing/debugging/iterating, but, none of them are "for keeps".

On my next commit on this PR I'll revert `src/Makevars.in` to what's in `main`.

Reply:
(And the long and the short of it is that we are supposed to put our local tweaks into the local file `~/.R/Makevars`, and that works.  It's just one of those things -- I'll show you next time we're both on coffee break.)